scripts/azureml-assets/azureml/assets/release/model_publish_utils/mlflow_model_utils.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import logging
+import os
+import azureml.evaluate.mlflow as mlflow
+from mlflow.models import ModelSignature
+from sqlalchemy import true
+from transformers import  AutoTokenizer, AutoConfig, pipeline
+from transformers import (
+    AutoModelForSequenceClassification,
+    AutoModelForMaskedLM,
+    AutoModelForTokenClassification,
+    AutoModelForQuestionAnswering,
+    AutoModelWithLMHead
+)
+
+class MLFlowModelUtils:
+
+    def __init__(self, name, task_name, flavor, model_dir):
+        self.name = name
+        self.task_name = task_name
+        self.model_dir = model_dir
+        self.flavor = flavor
+
+    def _convert_to_mlflow_hftransformers(self):
+        config = AutoConfig.from_pretrained(self.name)
+        misc_conf = {"task_type": self.task_name}
+        task_model_mapping = {
+            "multiclass": AutoModelForSequenceClassification,
+            "multilabel": AutoModelForSequenceClassification,
+            "fill-mask": AutoModelForMaskedLM,
+            "ner": AutoModelForTokenClassification,
+            "question-answering": AutoModelForQuestionAnswering,
+            "summarization": AutoModelWithLMHead,
+            "text-generation": AutoModelWithLMHead,
+            "text-classification": AutoModelForSequenceClassification
+        }
+        if self.task_name in task_model_mapping:
+            model = task_model_mapping[self.task_name].from_pretrained(self.name, config=config)
+        elif "translation" in self.task_name:
+            model = AutoModelWithLMHead.from_pretrained(self.name, config=config)
```

Comment:
@novaturient95 @ghyadav Is this the step where we saw `MemoryError` on the agents?

Reply:
yes. In case of disk space related errors, it fails while downloading/cloning model